greymatter-io/operator
# Remove Mesh CRD

[sc-16361]
[sc-19990]

https://github.com/greymatter-io/gitops-core/pull/42 must be merged first, and the operator's submodule must be updated, but see this comment before charging ahead: https://app.shortcut.com/grey-matter/story/19990/finish-removing-mesh-crd#activity-20273


### How to test

- Make sure the operator AND the submodule are both pointed to "remove-mesh-crd" for the BUILD.
- Point the submodule at this version of gitops-core: https://github.com/greymatter-io/gitops-core/pull/42
- Temporarily update the operator's StatefulSet in pkg/cuemodule/core/k8s/output/operator.cue with a `args:` flag `"-branch", "remove-mesh-crd"` to make the operator pull that^ branch for GitOps as well. (You'll need to make a temporary branch off of this branch for GitOps-related testing steps, so perhaps do _that_ now so you can change stuff freely with GitOps later).
- Build and deploy the operator code from this branch. Alternatively, this image has been built from this branch already: quay.io/greymatterio/operator:remove-mesh-crd
- Launch that operator image. This will do it:
```
(
cd pkg/cuemodule/core

kubectl create namespace gm-operator

kubectl create secret docker-registry gm-docker-secret \
  --docker-server=quay.io \
  --docker-username=$QUAY_USERNAME \
  --docker-password=$QUAY_PASSWORD \
  -n gm-operator

kubectl create secret generic greymatter-sync-secret \
  --from-file=id_ed25519=$HOME/.ssh/id_ed25519 \
  -n gm-operator

cue eval -c ./k8s/outputs --out text \
         -t operator_image=quay.io/greymatterio/operator:remove-mesh-crd \
         -e operator_manifests_yaml | kubectl apply -f -
)
```

- Confirm it deploys the mesh components
- Confirm GitOps live updates take, and end up stable
- Confirm sidecar injection/configuration succeeds
- Confirm deleting the operator’s ClusterRoleBinding) removes everything (plain delete of all operator stuff will do) (the Mesh CRD used to "own" everything, now the operator's ClusterRoleBinding owns everything, so if you delete it, many things should be cleaned up.
- Also confirm with spire turned on (re-deploy with that setting updated in the gitops repo - you'd probably want to branch off of the branch in gitops-core and change and push a temporary branch that the operator is pointed to for this)
- Confirm watch_namespace array can receive an appended customer namespace to which a workload can be deployed
